### PR TITLE
macos_userdefaults: Fix ruby 2.7 keyword args warning and refactor a bit

### LINKED ISSUE #9891: macos_userdefaults shell_out syntax deprecation warning
## Description
The macos_userdefaults produces a deprecation warning on its invocation of `shell_out`

## Chef Version
Reported from 16.1.x

## Client Output
```
* macos_userdefaults[...] action write
/opt/chef/embedded/lib/ruby/gems/2.7.0/gems/chef-16.1.0/lib/chef/resource/macos_userdefaults.rb:81:
warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
```

### PR DESCRIPTION
**Pull request by phiggins**
Fixes #9891

### CODE REVIEW

File: lib/chef/resource/macos_userdefaults.rb [hunk 1]
```diff
@@ -74,12 +74,17 @@ def coerce_booleans(val)
...
+        cmd << "'#{desired.key}' " if desired.key
+        cmd << " | grep -qx '#{value}'"
+
+        vc = if desired.user.nil?
+          shell_out(cmd)
+        else
+          shell_out(cmd, user: desired.user)
+        end
+
+        is_set vc.exitstatus == 0
```
Comment (tas50) [inline lib/chef/resource/macos_userdefaults.rb:87]: You should be able to just use .error? here

Comment (phiggins) [reply lib/chef/resource/macos_userdefaults.rb:87]: Good call!

File: lib/chef/resource/macos_userdefaults.rb [hunk 2]
```diff
@@ -74,12 +74,17 @@ def coerce_booleans(val)
...
+        cmd << "'#{desired.key}' " if desired.key
+        cmd << " | grep -qx '#{value}'"
+
+        vc = if desired.user.nil?
+          shell_out(cmd)
+        else
+          shell_out(cmd, user: desired.user)
+        end
+
+        is_set !vc.error?
```
Comment (tas50) [inline lib/chef/resource/macos_userdefaults.rb:87]: This is definitely an improvement. Long term we need to kill this is_set entirely.

**Comment by tas50** [comment]
@phiggins You have a minor chefstyle warning on this one

**Pull request merged**